File: AndroidSimple/src/main/java/com/pinger/repertory/widget/falling/FallingViewActivity.java
package com.pinger.repertory.widget.falling;

import com.pinger.baselib.base.BaseActivity;
import com.pinger.repertory.R;
import com.pinger.widget.view.falling.FallingEntity;
import com.pinger.widget.view.falling.FallingView;

/**
 * @author Pinger
 * @since 2018/1/11 0011 上午 11:51
 */

public class FallingViewActivity extends BaseActivity {
    @Override
    protected int getLayoutResID() {
        return R.layout.activity_falling_view;
    }

    @Override
    protected void initView() {
        setActionBar(getString(R.string.widget_view_falling_snow), true);

        FallingView fallingView = (FallingView) findViewById(R.id.falling_view);

        FallingEntity.Builder builder = new FallingEntity.Builder(getResources().getDrawable(R.mipmap.ic_snow));
        FallingEntity fallingEntity = builder
                .setSpeed(7, true)
                .setSize(50, 50, true)
                .setWind(5, true, true)
                .build();

        fallingView.addFallEntity(fallingEntity, 100);
    }
}

File: sample/src/main/java/com/fungo/repertory/widget/falling/FallingViewActivity.kt
package com.fungo.repertory.widget.falling

import android.support.v4.content.ContextCompat
import com.fungo.baselib.base.basic.BaseActivity
import com.fungo.baseuilib.view.falling.FallingEntity
import com.fungo.baseuilib.view.falling.FallingView
import com.fungo.repertory.R

/**
 * @author Pinger
 * @since 2018/1/11 0011 上午 11:51
 */

class FallingViewActivity : BaseActivity() {
    override val layoutResID: Int
        get() = R.layout.activity_falling_view

    override fun initView() {
        setToolBar(getString(R.string.widget_view_falling_snow), true)

        val fallingView = findViewById<FallingView>(R.id.falling_view)

        val builder = FallingEntity().Builder(ContextCompat.getDrawable(this, R.mipmap.ic_snow)!!)
        val fallingEntity = builder
                .setSpeed(7, true)
                .setSize(50, 50, true)
                .setWind(5, true, true)
                .build()

        fallingView.addFallEntity(fallingEntity, 100)
    }
}
